sdcm/provision/aws/capacity_reservation.py

```diff
@@ -33,7 +33,14 @@ class SCTCapacityReservation:
     def _get_cr_request_based_on_sct_config(params) -> Tuple[dict[str, int], int]:
         instance_counts = {}
         nemesis_node_count = params.get("nemesis_add_node_cnt") or 0
-        cluster_max_size = (params.get("cluster_target_size") or params.get("n_db_nodes")) + nemesis_node_count
+
+        cluster_max_size = (params.get("cluster_target_size") or params.get("n_db_nodes"))
+
+        if nemesis_grow_shrink_instance_type := params.get("nemesis_grow_shrink_instance_type"):
+            instance_counts[nemesis_grow_shrink_instance_type] = nemesis_node_count
```

Comment:
in unrealistic scenario when `nemesis_grow_shrink_instance_type` is the same like `instance_type_db` or like monitoring/loader node type then this cause problems. 
```
instance_counts[nemesis_grow_shrink_instance_type] = instance_counts.get(nemesis_grow_shrink_instance_type, 0) + nemesis_node_count
```

Reply:
we can guard that in the configuration itself, i.e. forbid using the same instance type as db as part of validation ?

Reply:
I think above proposal is easier and I think we don't have to restrict it then.

Reply:
why not use defaultdict with 0 ? and add to it as needed ?

Reply:
also can be done this way.

Reply:
it's looks much better now

I've test it with the unittests, but didn't want to complex them with add this logic into them